Portscout/SiteHandler/GitHub.pm

```diff
@@ -165,6 +138,41 @@ sub GetFiles
 }
 
 
+#------------------------------------------------------------------------------
+# Func: _call_github_api()
+# Desc: Calls the github api making use of settings.
+#
+# Args: $resource - Resource to query (e.g. "/repos/project/releases")
+#
+# Retn: Parsed JSON
+#------------------------------------------------------------------------------
+
+sub _call_github_api {
```

Comment:
This function encapsulates calls to github. It does GET requests to api.github.com, adds _github_token_ if one was configured and decodes the JSON format response.

Reply:
Nice!